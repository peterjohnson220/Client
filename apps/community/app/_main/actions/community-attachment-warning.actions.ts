import { Action } from '@ngrx/store';

export const OPEN_COMMUNITY_ATTACHMENTS_WARNING_MODAL = '[Community/Attachments] Open Community Attachments Warning Modal';
export const CLOSE_COMMUNITY_ATTACHMENTS_WARNING_MODAL = '[Community/Attachments] Close Community Attachments Warning Modal';

export class OpenCommunityAttachmentsWarningModal implements Action {
  readonly type = OPEN_COMMUNITY_ATTACHMENTS_WARNING_MODAL;
  constructor(public payload: string) {}
}

export class CloseCommunityAttachmentsWarningModal implements Action {
  readonly type = CLOSE_COMMUNITY_ATTACHMENTS_WARNING_MODAL;
  constructor() {}
}

export type Actions
  = OpenCommunityAttachmentsWarningModal
  | CloseCommunityAttachmentsWarningModal;
