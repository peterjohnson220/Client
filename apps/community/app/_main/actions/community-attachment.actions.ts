import { Action } from '@ngrx/store';
import { CommunityAttachmentModalState } from 'libs/models';

export const OPEN_COMMUNITY_ATTACHMENTS_MODAL = '[Community/Attachments] Open Community Attachments Modal';
export const CLOSE_COMMUNITY_ATTACHMENTS_MODAL = '[Community/Attachments] Close Community Attachments Modal';
export const SAVE_COMMUNITY_ATTACHMENTS_STATE = '[Community/Attachments] Save Community Attachments State';
export const CLEAR_COMMUNITY_ATTACHMENTS_STATE = '[Community/Attachments] Clear Community Attachments State';

export class OpenCommunityAttachmentsModal implements Action {
  readonly type = OPEN_COMMUNITY_ATTACHMENTS_MODAL;
  constructor(public payload: string) {}
}

export class CloseCommunityAttachmentsModal implements Action {
  readonly type = CLOSE_COMMUNITY_ATTACHMENTS_MODAL;
  constructor(public payload: string) {}
}

export class SaveCommunityAttachmentsState implements Action {
  readonly type = SAVE_COMMUNITY_ATTACHMENTS_STATE;
  constructor(public payload: CommunityAttachmentModalState) {}
}

export class ClearCommunityAttachmentsState implements Action {
  readonly type = CLEAR_COMMUNITY_ATTACHMENTS_STATE;
  constructor(public payload: string) {}
}

export type Actions
  = OpenCommunityAttachmentsModal
  | CloseCommunityAttachmentsModal
  | SaveCommunityAttachmentsState
  | ClearCommunityAttachmentsState;
