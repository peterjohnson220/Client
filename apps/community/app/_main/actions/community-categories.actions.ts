import { Action } from '@ngrx/store';
import { CommunityCategory } from 'libs/models/community/community-categories.model';

export const GETTING_COMMUNITY_CATEGORIES = '[Community/Categories] Get Community Categories';
export const GETTING_COMMUNITY_CATEGORIES_SUCCESS = '[Community/Categories] Get Community Categories Success';
export const GETTING_COMMUNITY_CATEGORIES_ERROR = '[Community/Categories] Get Community Categories Error';
export const ADDING_COMMUNITY_POST_TO_CATEGORIES_COUNT = '[Community/Categories] Add To Community Categories Count';
export const SUBTRACTING_COMMUNITY_POST_TO_CATEGORIES_COUNT = '[Community/Categories] Subtract To Community Categories Count';

export class GettingCommunityCategories implements Action {
  readonly type = GETTING_COMMUNITY_CATEGORIES;
  constructor() {}
}

export class GettingCommunityCategoriesSuccess implements Action {
  readonly type = GETTING_COMMUNITY_CATEGORIES_SUCCESS;
  constructor(public payload: CommunityCategory[]) {}
}

export class GettingCommunityCategoriesError implements Action {
  readonly type = GETTING_COMMUNITY_CATEGORIES_ERROR;
  constructor() {}
}

export class AddingCommunityPostToCategoriesCount implements Action {
  readonly type = ADDING_COMMUNITY_POST_TO_CATEGORIES_COUNT;
  constructor(public payload: any) {}
}

export class SubtractingCommunityPostToCategoriesCount implements Action {
  readonly type = SUBTRACTING_COMMUNITY_POST_TO_CATEGORIES_COUNT;
  constructor(public payload: any) {}
}

export type Actions
  =  GettingCommunityCategories
  | GettingCommunityCategoriesSuccess
  | GettingCommunityCategoriesError
  | AddingCommunityPostToCategoriesCount
  | SubtractingCommunityPostToCategoriesCount;
