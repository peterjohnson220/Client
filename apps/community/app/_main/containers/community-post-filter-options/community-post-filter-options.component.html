<div *ngIf="filters$ | async as filter" class="filter-wrapper pt-1">
  <span *ngIf="filter.PostIds.length > 0 || filter.ReplyIds.length > 0">
     <button (click)="buttonViewAllClicked()" type="button" class="btn btn-secondary submit-button-container">
        <i class="far fa-eye pr-2"></i>View All Posts
      </button>
  </span>
  <span *ngIf="filter.TagFilter.Tags.length > 0 || filter.CategoryFilter.Category.length > 0" class="filter-text font-weight-bold">Showing content with:</span>
  <ng-container *ngFor="let category of filter.CategoryFilter.Category;">

    <span class="d-inline-block m-1 filter-container p-1">
      <button class="m-0 p-0 border-0 d-inline-block font-weight-bold" (click)="buttonClicked('category', category)">
        <span class="pr-1">{{ category }}</span>
        <i class="fas fa-times-circle icon"></i>
      </button>
    </span>

  </ng-container>
  <ng-container *ngFor="let tag of filter.TagFilter.Tags;">

    <span class="d-inline-block m-1 filter-container p-1">
      <button class="m-0 p-0 border-0 d-inline-block font-weight-bold" (click)="buttonClicked('tag', tag)">
        <span class="pr-1">{{ tag.TagName }}</span>
        <i class="fas fa-times-circle icon"></i>
      </button>
    </span>

  </ng-container>
  <hr *ngIf="filter.TagFilter.Tags.length > 0 || filter.CategoryFilter.Category.length > 0 ||
    filter.PostIds.length > 0 || filter.ReplyIds.length > 0" >
</div>
