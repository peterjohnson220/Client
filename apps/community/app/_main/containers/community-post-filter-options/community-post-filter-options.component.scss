.filter-wrapper
{
  background-color: white;
  position: sticky;
  top: -10px;
  z-index: 999;
}

.filter-container {
  background-color: #ECF0F5;
  color: #515151;
  white-space: normal;
}

.filter-container button {
  background: none;
  color: #515151;
}

.filter-text
{
  color: #51667E
}

.icon {
  cursor: pointer;
}

.submit-button-container{
  color: cornflowerblue;
}
