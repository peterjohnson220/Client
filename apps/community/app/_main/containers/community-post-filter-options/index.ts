export * from './community-post-filter-options.component';
