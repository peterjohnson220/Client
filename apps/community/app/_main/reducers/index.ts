import { createSelector, createFeatureSelector } from '@ngrx/store';

import * as fromRoot from 'libs/state/state';
import * as cloneDeep from 'lodash.clonedeep';

import * as fromCommunityPollRequestReducer from './community-poll-request.reducer';
import * as fromCommunityPollResponseReducer from './community-poll-response.reducer';
import * as fromCommunityPostReducer from './community-post.reducer';
import * as fromCommunityPostReplyReducer from './community-post-reply.reducer';
import * as fromCommunityPostAddReplyViewReducer from './community-post-add-reply-view.reducer';
import * as fromCommunityPostFilterReplyViewReducer from './community-post-filter-reply-view.reducer';
import * as fromCommunityTagReducer from './community-tag.reducer';
import * as fromCommunityJobReducer from './community-job.reducer';
import * as fromCommunityCategoriesReducer from './community-categories.reducer';
import * as fromCommunityPostFilterOptionsReducer from './community-post-filter-options.reducer';
import * as fromCommunityLikeReducer from './community-like.reducer';
import * as fromCommunitySearchReducer from './community-search.reducer';
import * as fromCommunityIndustryReducer from './community-industry.reducer';
import * as fromCommunityCompanySizeReducer from './community-company-size.reducer';
import * as fromCommunityTopicReducer from './community-topic.reducer';
import { CommunityConstants } from '../models';
import { populatePostReplies } from '../helpers/model-mapping.helper';

// Feature area state
export interface CommunityState {
  communityPollRequest: fromCommunityPollRequestReducer.State;
  communityPollResponse: fromCommunityPollResponseReducer.State;
  communityPost: fromCommunityPostReducer.State;
  communityPostReply: fromCommunityPostReplyReducer.State;
  communityPostFilteredReplyView: fromCommunityPostFilterReplyViewReducer.State;
  communityPostAddReplyView: fromCommunityPostAddReplyViewReducer.State;
  communityTags: fromCommunityTagReducer.State;
  communityJob: fromCommunityJobReducer.State;
  communityCategories: fromCommunityCategoriesReducer.State;
  communityPostFilterOptions: fromCommunityPostFilterOptionsReducer.State;
  communityLikes: fromCommunityLikeReducer.State;
  communitySearch: fromCommunitySearchReducer.State;
  communityIndustries: fromCommunityIndustryReducer.State;
  communityCompanySizes: fromCommunityCompanySizeReducer.State;
  communityTopic: fromCommunityTopicReducer.State;
}

// Extend root state with feature area state
export interface State extends fromRoot.State {
  communityPoll: CommunityState;
}

// Feature area reducers
export const reducers = {
  communityPollRequest: fromCommunityPollRequestReducer.reducer,
  communityPollResponse: fromCommunityPollResponseReducer.reducer,
  communityPost: fromCommunityPostReducer.reducer,
  communityPostReply: fromCommunityPostReplyReducer.reducer,
  communityPostFilteredReplyView: fromCommunityPostFilterReplyViewReducer.reducer,
  communityPostAddReplyView: fromCommunityPostAddReplyViewReducer.reducer,
  communityTags: fromCommunityTagReducer.reducer,
  communityJob: fromCommunityJobReducer.reducer,
  communityCategories: fromCommunityCategoriesReducer.reducer,
  communityPostFilterOptions: fromCommunityPostFilterOptionsReducer.reducer,
  communityLikes: fromCommunityLikeReducer.reducer,
  communitySearch: fromCommunitySearchReducer.reducer,
  communityIndustries: fromCommunityIndustryReducer.reducer,
  communityCompanySizes: fromCommunityCompanySizeReducer.reducer,
  communityTopic: fromCommunityTopicReducer.reducer
};

// select feature area
export const selectCommunityState = createFeatureSelector<CommunityState>('community');

// Feature selectors
export const selectFromCommunityPollRequestState = createSelector(
  selectCommunityState,
  (state: CommunityState) => state.communityPollRequest
);

export const selectFromCommunityPollResponseState = createSelector(
  selectCommunityState,
  (state: CommunityState) => state.communityPollResponse
);

export const selectFromCommunityPostState = createSelector(
  selectCommunityState,
  (state: CommunityState) => state.communityPost
);

export const selectFromCommunityPostReplyState = createSelector(
  selectCommunityState,
  (state: CommunityState) => state.communityPostReply
);

export const selectFromCommunityPostAddReplyViewState = createSelector(
  selectCommunityState,
  (state: CommunityState) => state.communityPostAddReplyView
);

export const selectFromCommunityPostFilterReplyViewState = createSelector(
  selectCommunityState,
  (state: CommunityState) => state.communityPostFilteredReplyView
);

export const selectFromCommunityTagState = createSelector(
  selectCommunityState,
  (state: CommunityState) => state.communityTags
);

export const selectFromCommunityJobState =  createSelector(
  selectCommunityState,
  (state: CommunityState) => state.communityJob
);

export const selectFromCommunityCategoriesState =  createSelector(
  selectCommunityState,
  (state: CommunityState) => state.communityCategories
);

export const selectFromCommunityPostFilterOptionsState =  createSelector(
  selectCommunityState,
  (state: CommunityState) => state.communityPostFilterOptions
  );

export const selectFromCommunityLikeState = createSelector(
    selectCommunityState,
    (state: CommunityState) => state.communityLikes
  );

export const selectFromCommunitySearchState = createSelector(
  selectCommunityState,
  (state: CommunityState) => state.communitySearch
);


export const selectFromCommunityIndustryState = createSelector(
  selectCommunityState,
  (state: CommunityState) => state.communityIndustries
  );

export const selectFromCommunityTopicState = createSelector(
  selectCommunityState,
  (state: CommunityState) => state.communityTopic
);

export const selectFromCommunityCompanySizeState = createSelector(
  selectCommunityState,
  (state: CommunityState) => state.communityCompanySizes
);

// Community Poll Selectors
export const getGettingCommunityPollRequests = createSelector(
  selectFromCommunityPollRequestState,
  fromCommunityPollRequestReducer.getGettingCommunityPollRequests
);

export const getCommunityPollRequests = createSelector(
  selectFromCommunityPollRequestState,
  fromCommunityPollRequestReducer.getCommunityPollRequests
);

export const getGettingCommunityPollRequestsError = createSelector(
  selectFromCommunityPollRequestState,
  fromCommunityPollRequestReducer.getGettingCommunityPollRequestsError
);

export const getGettingCommunityPollRequestsLoaded = createSelector(
  selectFromCommunityPollRequestState,
  fromCommunityPollRequestReducer.getGettingCommunityPollRequestsLoaded
);

export const getSubmittingCommunityPollRequestResponses = createSelector(
  selectFromCommunityPollRequestState,
  fromCommunityPollRequestReducer.getSubmittingCommunityPollRequestResponses
);

export const getSubmittingCommunityPollRequestResponse = createSelector(
  selectFromCommunityPollRequestState,
  fromCommunityPollRequestReducer.getSubmittingCommunityPollRequestResponse
);

export const getGettingCommunityPollResponses = createSelector(
  selectFromCommunityPollResponseState,
  fromCommunityPollResponseReducer.getGettingCommunityPollResponses
);

export const getGettingCommunityPollResponsesSuccess = createSelector(
  selectFromCommunityPollResponseState,
  fromCommunityPollResponseReducer.getGettingCommunityPollResponsesSuccess
);

export const getGettingCommunityPollResponsesError = createSelector(
  selectFromCommunityPollResponseState,
  fromCommunityPollResponseReducer.getGettingCommunityPollResponsesError
);

export const getDismissingCommunityPollResponse = createSelector(
  selectFromCommunityPollResponseState,
  fromCommunityPollResponseReducer.getDismissingCommunityPollResponse
);

export const getDismissingCommunityPollResponseSuccess = createSelector(
  selectFromCommunityPollResponseState,
  fromCommunityPollResponseReducer.getDismissingCommunityPollResponseSuccess
);

export const getDismissingCommunityPollResponseError = createSelector(
  selectFromCommunityPollResponseState,
  fromCommunityPollResponseReducer.getDismissingCommunityPollResponseError
);

export const getDismissedResponses = createSelector(
  selectFromCommunityPollResponseState,
  fromCommunityPollResponseReducer.getDismissedResponses
);
export const hasNonDismissedResponses = createSelector(
  getDismissedResponses,
  getGettingCommunityPollResponsesSuccess,
  (dismissedResponses, allResponses) => {
    if (dismissedResponses.length > 0) {
      const filteredResponses = allResponses.filter((response) => dismissedResponses.indexOf(response.CommunityPollId) === -1);
      return filteredResponses.length > 0;
    } else {
      return allResponses.length > 0;
    }
  }
);
// Community Post Selectors
export const {
  selectAll: getCommunityPosts,
} = fromCommunityPostReducer.adapter.getSelectors(selectFromCommunityPostState);

export const getSubmittingCommunityPosts = createSelector(
  selectFromCommunityPostState,
  fromCommunityPostReducer.getSubmittingCommunityPosts
);

export const getSubmittingCommunityPostsError = createSelector(
  selectFromCommunityPostState,
  fromCommunityPostReducer.getSubmittingCommunityPostsError
);

export const getSubmittingCommunityPostsSuccess = createSelector(
  selectFromCommunityPostState,
  fromCommunityPostReducer.getSubmittingCommunityPostsSuccess
);

export const getGettingCommunityPosts = createSelector(
  selectFromCommunityPostState,
  fromCommunityPostReducer.getGettingCommunityPosts
);

export const getGettingCommunityPostsError = createSelector(
  selectFromCommunityPostState,
  fromCommunityPostReducer.getGettingCommunityPostsError
);

export const getAddingCommunityDiscussionPoll = createSelector(
  selectFromCommunityPostState,
  fromCommunityPostReducer.getAddingCommunityDiscussionPoll
);

export const getAddingCommunityDiscussionPollSuccess = createSelector(
  selectFromCommunityPostState,
  fromCommunityPostReducer.getAddingCommunityDiscussionPollSuccess
);

export const getAddingCommunityDiscussionPollError = createSelector(
  selectFromCommunityPostState,
  fromCommunityPostReducer.getAddingCommunityDiscussionPollError
);

export const getDiscussionPagingOptions = createSelector(
  selectFromCommunityPostState,
  fromCommunityPostReducer.getDiscussionPagingOptions
);

export const getTotalDiscussionResultsOnServer = createSelector(
  selectFromCommunityPostState,
  fromCommunityPostReducer.getTotalResultsOnServer
);

export const getLoadingNextBatchPosts = createSelector(
  selectFromCommunityPostState,
  fromCommunityPostReducer.getLoadingNextBatchPosts
);

export const getLoadingPreviousBatchPosts = createSelector(
  selectFromCommunityPostState,
  fromCommunityPostReducer.getLoadingPreviousBatchPosts
);

export const getMaximumReplies = createSelector(
  selectFromCommunityPostState,
  fromCommunityPostReducer.getMaximumReplies
);

export const getLoadingCommunityPost = createSelector(
  selectFromCommunityPostState,
  fromCommunityPostReducer.getLoadingCommunityPost
);

export const getLoadingCommunityPostSuccess = createSelector(
  selectFromCommunityPostState,
  fromCommunityPostReducer.getLoadingCommunityPostSuccess
);

export const getLoadingCommunityPostError = createSelector(
  selectFromCommunityPostState,
  fromCommunityPostReducer.getLoadingCommunityPostError
);

export const getHasNextBatchPostsOnServer = createSelector(
  getTotalDiscussionResultsOnServer,
  selectFromCommunityPostState,
  (totalOnServer, pagingOptions) => {
    return totalOnServer > CommunityConstants.POSTS_PER_BATCH * pagingOptions.endIndexDisplayed;
  }
);

export const getHasPreviousBatchPostsOnServer = createSelector(
  selectFromCommunityPostState,
  (pagingOptions) => {
    return pagingOptions.startIndexDisplayed > 1;
  }
);

export const getCommunityPostDeleted = createSelector(
  selectFromCommunityPostState,
  fromCommunityPostReducer.getCommunityPostDeleted
);

// Community Post Reply Selectors

export const {
  selectAll: getCommunityPostReply,
  selectEntities: getCommunityPostReplyEntities,
} = fromCommunityPostReplyReducer.adapter.getSelectors(selectFromCommunityPostReplyState);

export const getGettingCommunityPostReplies = createSelector(
  selectFromCommunityPostReplyState,
  fromCommunityPostReplyReducer.getGettingCommunityPostReplies
);

export const getGettingCommunityPostRepliesError = createSelector(
  selectFromCommunityPostReplyState,
  fromCommunityPostReplyReducer.getGettingCommunityPostRepliesError
);

export const getAddingCommunityPostReply = createSelector(
  selectFromCommunityPostReplyState,
  fromCommunityPostReplyReducer.getAddingCommunityPostReply
);

export const getAddingCommunityPostReplyError = createSelector(
  selectFromCommunityPostReplyState,
  fromCommunityPostReplyReducer.getAddingCommunityPostReplyError
);

export const getAddingCommunityPostReplySuccess = createSelector(
  selectFromCommunityPostReplyState,
  fromCommunityPostReplyReducer.getAddingCommunityPostReplySuccess
);

// Community Post Add Reply View selector
export const getCommunityPostAddReplyView = createSelector(
  selectFromCommunityPostAddReplyViewState,
  fromCommunityPostAddReplyViewReducer.getCommunityPostAddReplyView
);

export const getFilteredCommunityPostAddReplyView = createSelector(
  getCommunityPostReplyEntities,
  getCommunityPostAddReplyView,
  (replies, addReplies) => addReplies.reduce((acc, id) => {
    return replies[ id ] ? [ ...acc, replies[ id ] ] : acc;
  }, [])
);

// Community Post Filtered Reply View selector
export const getCommunityPostFilterReplyView = createSelector(
  selectFromCommunityPostFilterReplyViewState,
  fromCommunityPostFilterReplyViewReducer.getCommunityPostFilterReplyView
);

export const getFilteredCommunityPostReplyView = createSelector(
  getCommunityPostReplyEntities,
  getCommunityPostFilterReplyView,
  (replies, filterReplies) => filterReplies.reduce((acc, id) => {
    return replies[ id ] ? [ ...acc, replies[ id ] ] : acc;
  }, [])
);

// Community Posts With Replies
export const getCommunityPostsCombinedWithReplies = createSelector(
  getCommunityPosts,
  getCommunityPostReplyEntities,
  getCommunityPostAddReplyView,
  getCommunityPostFilterReplyView,
  (posts, replies, addReplies, filterReplies) => {
    if (posts && replies) {
      const postlist = cloneDeep(posts);
      postlist.forEach(post => {
        populatePostReplies(post, addReplies, filterReplies, replies);
      });
      return postlist;
    } else {
      return posts;
    }
  }
);


// Community Post With Replies
export const getCommunityPostCombinedWithReplies = createSelector(
  getCommunityPosts,
  getCommunityPostReplyEntities,
  getCommunityPostAddReplyView,
  getLoadingCommunityPostSuccess,
  (posts,  replies, addReplies, postId) => {
    const post =  postId ? posts.find(x => x.Id === postId) : null;
    if (!post) {
      return post;
    }

    const postWithReplies = cloneDeep(post);
    populatePostReplies(postWithReplies, addReplies, [], replies);
    return postWithReplies;
  }
);

// Community Tag Selectors

export const getSuggestingCommunityTags = createSelector(
  selectFromCommunityTagState,
  fromCommunityTagReducer.getSuggestingCommunityTags
);

export const getCommunityTags = createSelector(
  selectFromCommunityTagState,
  fromCommunityTagReducer.getCommunityTags
);

export const getSuggestingCommunityTagsPostId = createSelector(
  selectFromCommunityTagState,
  fromCommunityTagReducer.getCommunityTagsPostId
);

export const getSuggestingCommunityTagsError = createSelector(
  selectFromCommunityTagState,
  fromCommunityTagReducer.getSuggestingCommunityTagsError
);

// Community Job Selectors
export const {
  selectAll: getCommunityJobs,
} = fromCommunityJobReducer.adapter.getSelectors(selectFromCommunityJobState);

export const getSubmittingCommunityJobs = createSelector(
  selectFromCommunityJobState,
  fromCommunityJobReducer.getSubmittingCommunityJobs
);

export const getSubmittingCommunityJobsError = createSelector(
  selectFromCommunityJobState,
  fromCommunityJobReducer.getSubmittingCommunityJobsError
);

export const getSubmittingCommunityJobsSuccess = createSelector(
  selectFromCommunityJobState,
  fromCommunityJobReducer.getSubmittingCommunityJobsSuccess
);

export const getGettingCommunityJobs = createSelector(
  selectFromCommunityJobState,
  fromCommunityJobReducer.getGettingCommunityJobs
);

export const getGettingCommunityJobsError = createSelector(
  selectFromCommunityJobState,
  fromCommunityJobReducer.getGettingCommunityJobsError
);

export const getLoadingMoreResults = createSelector(
  selectFromCommunityJobState,
  fromCommunityJobReducer.getLoadingMoreResults
);

export const getPagingOptions = createSelector(
  selectFromCommunityJobState,
  fromCommunityJobReducer.getPagingOptions
);

export const getTotalResultsOnServer = createSelector(
  selectFromCommunityJobState,
  fromCommunityJobReducer.getTotalResultsOnServer
);

export const getHasMoreResultsOnServer = createSelector(
  getCommunityJobs,
  getTotalResultsOnServer,
  (jobs, totalOnServer) => {
    return totalOnServer > jobs.length;
  }
);

// Community Categories Selectors

export const {
  selectAll: getCommunityCategories,
} = fromCommunityCategoriesReducer.adapter.getSelectors(selectFromCommunityCategoriesState);

export const getGettingCommunityCategories = createSelector(
  selectFromCommunityCategoriesState,
  fromCommunityCategoriesReducer.getGettingCommunityCategories
);

export const getGettingCommunityCategoriesError = createSelector(
  selectFromCommunityCategoriesState,
  fromCommunityCategoriesReducer.getGettingCommunityCategoriesError
);

export const getCommunityPostFilterOptions = createSelector(
  selectFromCommunityPostFilterOptionsState,
  fromCommunityPostFilterOptionsReducer.getCommunityPostFilterOptions
);

export const getFilteredByPost = createSelector(
  selectFromCommunityPostFilterOptionsState,
  fromCommunityPostFilterOptionsReducer.getFilteredByPost
);

// Community Like Selectors
export const getLoadingCommunityLikes = createSelector(
  selectFromCommunityLikeState,
  fromCommunityLikeReducer.getLoadingCommunityLikes
);

export const getLoadingCommunityLikesSuccess = createSelector(
  selectFromCommunityLikeState,
  fromCommunityLikeReducer.getLoadingCommunityLikesSuccess
);

export const getLoadingCommunityLikesError = createSelector(
  selectFromCommunityLikeState,
  fromCommunityLikeReducer.getLoadingCommunityLikesError
);

export const getFilterTitle = createSelector(
  getFilteredByPost,
  getCommunityPostsCombinedWithReplies,
  (filteredByPost, postsWithReplies) => {
    if (filteredByPost && postsWithReplies.length === 1) {
      const discussionType = postsWithReplies[0].UserPollRequest == null ? 'post' : 'poll';

      return `${postsWithReplies[0].UserInfo.UserFirstName} ${postsWithReplies[0].UserInfo.UserLastName}'s ${discussionType} `;
    }
  }
);

// Community Search Selectors
export const {
  selectAll: getCommunitySearchResults,
} = fromCommunitySearchReducer.adapter.getSelectors(selectFromCommunitySearchState);

export const getLoadingSearchResults = createSelector(
  selectFromCommunitySearchState,
  fromCommunitySearchReducer.getLoadingSearchResults
);

export const getLoadingSearchResultsError = createSelector(
  selectFromCommunitySearchState,
  fromCommunitySearchReducer.getLoadingSearchResultsError
);

export const getCommunityLoadingMoreSearchResults = createSelector(
  selectFromCommunitySearchState,
  fromCommunitySearchReducer.getLoadingMoreSearchResults
);

export const getSearchResultsPagingOptions = createSelector(
  selectFromCommunitySearchState,
  fromCommunitySearchReducer.getSearchResultsPagingOptions
);

export const getCommunitySearchResultModal = createSelector(
  selectFromCommunitySearchState,
  fromCommunitySearchReducer.getSearchResultModalPostId
);

export const getTotalSearchResultsOnServer = createSelector(
  selectFromCommunitySearchState,
  fromCommunitySearchReducer.getTotalSearchResultsOnServer
);

export const getHasMoreSearchResultsOnServer = createSelector(
  getCommunitySearchResults,
  getTotalSearchResultsOnServer,
  (results, totalResults) => {
    return totalResults > results.length;
  }
);

// Community Industry Selectors
export const getLoadingCommunityIndustries = createSelector(
  selectFromCommunityIndustryState,
  fromCommunityIndustryReducer.getLoadingCommunityIndustries
);

export const getCommunityIndustries = createSelector(
  selectFromCommunityIndustryState,
  fromCommunityIndustryReducer.getCommunityIndustries
);

export const getLoadingCommunityIndustriesError = createSelector(
  selectFromCommunityIndustryState,
  fromCommunityIndustryReducer.getLoadingCommunityIndustriesError
);

// Community Company Size Selectors
export const getLoadingCommunityCompanySizes = createSelector(
  selectFromCommunityCompanySizeState,
  fromCommunityCompanySizeReducer.getLoadingCommunityCompanySizes
);

export const getCommunityCompanySizes = createSelector(
  selectFromCommunityCompanySizeState,
  fromCommunityCompanySizeReducer.getCommunityCompanySizes
);

export const getLoadingCommunityCompanySizesError = createSelector(
  selectFromCommunityCompanySizeState,
  fromCommunityCompanySizeReducer.getLoadingCommunityCompanySizesError
);

// Community Topic Selectors
export const getLoadingTopics = createSelector(
  selectFromCommunityTopicState,
  fromCommunityTopicReducer.getLoadingCommunityTopics
);

export const getLoadingTopicsError = createSelector(
  selectFromCommunityTopicState,
  fromCommunityTopicReducer.getLoadingCommunityTopicsError
);

export const getTopics = createSelector(
  selectFromCommunityTopicState,
  fromCommunityTopicReducer.getCommunityTopics
);
