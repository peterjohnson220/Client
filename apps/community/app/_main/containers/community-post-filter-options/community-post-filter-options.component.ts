import { Component } from '@angular/core';
import { Router } from '@angular/router';

import { Store } from '@ngrx/store';
import { Observable } from 'rxjs';

import * as fromCommunityPostReducer from '../../reducers';
import * as fromCommunityPostFilterOptionsReducer from '../../reducers';
import * as fromCommunityPostFilterOptionsActions from '../../actions/community-post-filter-options.actions';
import { FilterOptions } from '../../models/filter-options.model';

@Component({
  selector: 'pf-community-post-filter-options',
  templateUrl: './community-post-filter-options.component.html',
  styleUrls: ['./community-post-filter-options.component.scss'],
})
export class CommunityPostFilterOptionsComponent {
  filters$: Observable<FilterOptions>;

  constructor(public router: Router,
              public filterStore: Store<fromCommunityPostFilterOptionsReducer.State>,
              public store: Store<fromCommunityPostReducer.State>) {
    this.filters$ = this.filterStore.select(fromCommunityPostFilterOptionsReducer.getCommunityPostFilterOptions);
  }
  buttonClicked(type: string, item: any) {
    switch (type) {
      case 'tag':
        this.filterStore.dispatch(new fromCommunityPostFilterOptionsActions.DeletingCommunityTagFromFilterOptions(item));
        break;

      case 'category':
        this.filterStore.dispatch(new fromCommunityPostFilterOptionsActions.DeletingCommunityCategoryFromFilterOptions(item));
        break;
    }
  }

  buttonViewAllClicked() {
    this.filterStore.dispatch(new fromCommunityPostFilterOptionsActions.DeletingAllFilterOptions());
    this.router.navigateByUrl('/dashboard');
  }
}
