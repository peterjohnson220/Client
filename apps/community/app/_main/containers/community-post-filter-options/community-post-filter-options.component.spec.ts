import { NO_ERRORS_SCHEMA } from '@angular/core';
import { ReactiveFormsModule } from '@angular/forms';
import { TestBed, ComponentFixture } from '@angular/core/testing';
import { Router } from '@angular/router';
import { combineReducers, Store, StoreModule } from '@ngrx/store';

import * as fromRootState from 'libs/state/state';
import * as fromCommunityPostFilterOptionsReducer from '../../reducers';
import * as fromCommunityPostFilterOptionsActions from '../../actions/community-post-filter-options.actions';
import { CommunityPostFilterOptionsComponent } from './community-post-filter-options.component';


describe('CommunityCategoriesComponent', () => {
  let fixture: ComponentFixture<CommunityPostFilterOptionsComponent>;
  let instance: CommunityPostFilterOptionsComponent;
  let store: Store<fromRootState.State>;
  let router: Router;
  // Configure Testing Module for before each test
  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [
        StoreModule.forRoot({
          ...fromRootState.reducers,
          communityPostFilterOptions: combineReducers(fromCommunityPostFilterOptionsReducer.reducers)
        }),
        ReactiveFormsModule
      ],
      declarations: [
        CommunityPostFilterOptionsComponent
      ],
      providers: [
        {
          provide: Router,
          useValue: { navigate: jest.fn() },
        }
      ],
      // Shallow Testing
      schemas: [ NO_ERRORS_SCHEMA ]
    });

    store = TestBed.get(Store);
    router = TestBed.get(Router);

    spyOn(store, 'dispatch');

    fixture = TestBed.createComponent(CommunityPostFilterOptionsComponent);
    instance = fixture.componentInstance;
  });

  it('should show community categories', () => {
    expect(fixture).toMatchSnapshot();
  });
  it('should dispatch delete tag when button clicked', () => {
    const item = {};
    const expectedAction = new fromCommunityPostFilterOptionsActions.DeletingCommunityTagFromFilterOptions(item);
    const type = 'tag';
    instance.buttonClicked(type, item);

    expect(store.dispatch).toHaveBeenCalledWith(expectedAction);
  });
  it('should dispatch delete category when button clicked', () => {
    const item = {};
    const expectedAction = new fromCommunityPostFilterOptionsActions.DeletingCommunityCategoryFromFilterOptions(item);
    const type = 'category';
    instance.buttonClicked(type, item);

    expect(store.dispatch).toHaveBeenCalledWith(expectedAction);
  });
});
