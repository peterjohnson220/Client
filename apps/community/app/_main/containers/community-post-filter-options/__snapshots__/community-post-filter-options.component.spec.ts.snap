// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommunityCategoriesComponent should show community categories 1`] = `
<pf-community-post-filter-options
  filterStore={[Function Store]}
  filters$={[Function Store]}
  router={[Function Object]}
  store={[Function Store]}
>
  
</pf-community-post-filter-options>
`;
